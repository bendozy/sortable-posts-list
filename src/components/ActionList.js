import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Action from './Action';

export const ActionList = ({ actions }) => (
  <div className="ActionList">
    <div className="bg-white rounded shadow-md overflow-hidden">
      <div className="font-semibold bg-white p-4 text-gray-800 text-xl">List of actions commited</div>
      <div className="bg-gray-300 p-6 border border-gray-300 overflow-hidden shadow-md">
        {actions.map((action, index) => <Action key={action.key} action={action} index={index} />)}
        {actions.length === 0 && (
          <div className="bg-white p-4 rounded shadow-md flex items-center text-gray-500">
            <span>No actions commited</span>
          </div>
        )}
      </div>
    </div>
  </div>
);

ActionList.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export const mapStateToProps = ({ actions }) => ({ actions });

export default connect(mapStateToProps)(ActionList);
