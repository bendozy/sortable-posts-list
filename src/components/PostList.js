import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Post from './Post';
import { fetchPosts } from '../actions/posts';

import spinner from '../assets/Spinner.svg';

import '../styles/PostList.css';

export const PostList = (props) => {
  const { posts: { isLoading, posts } } = props;
  

  useEffect(() => {
    props.fetchPosts();
  }, []);

  const postCards = posts.map((post) => (
    <Post key={post.id} post={post} lastIndex={posts.length - 1} />
  ));

  return (
    <div className="PostList">
      <h1 className="mb-6 font-bold text-3xl text-white">Sortable Post List</h1>
      {isLoading && <img src={spinner} className="PostList-spinner" alt="loading-logo" />}
      {!isLoading && postCards}
    </div>
  );
};

PostList.propTypes = {
  fetchPosts: PropTypes.func.isRequired,
  posts: PropTypes.shape({
    isLoading: PropTypes.bool.isRequired,
    posts: PropTypes.arrayOf(PropTypes.object),
    order: PropTypes.arrayOf(PropTypes.number),
  }).isRequired,
};

export const mapStateToProps = ({ posts }) => ({ posts });

const mapDispatchToProps = {
  fetchPosts,
};

export default connect(mapStateToProps, mapDispatchToProps)(PostList);
