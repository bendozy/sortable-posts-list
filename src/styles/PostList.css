.PostList-spinner {
  min-height: 150px;
  background: #fff;
  margin: 0 auto;
  position: relative;
}
