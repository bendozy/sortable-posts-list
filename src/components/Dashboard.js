import React from 'react';
import PostList from './PostList';
import ActionList from './ActionList';

const Dashboard = () => (
  <div className="Dashboard container flex flex-wrap my-10 px-5">
    <div className="w-full md:w-1/2 px-8">
      <PostList />
    </div>
    <div className="w-full md:w-1/2 px-8">
      <ActionList />
    </div>
  </div>
);

export default Dashboard;
