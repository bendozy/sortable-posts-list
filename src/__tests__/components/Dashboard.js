import React from 'react';
import toJson from 'enzyme-to-json';
import Dashboard from '../../components/Dashboard';
import PostList from '../../components/PostList';
import ActionList from '../../components/ActionList';

describe('<App />', () => {
  it('should render a <App /> component', () => {
    const wrapper = shallow(<Dashboard />);

    expect(wrapper.find(PostList)).toHaveLength(1);
    expect(wrapper.find(ActionList)).toHaveLength(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
