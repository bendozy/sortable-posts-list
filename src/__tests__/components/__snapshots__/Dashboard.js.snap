// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render a <App /> component 1`] = `
<div
  className="Dashboard container flex flex-wrap my-10 px-5"
>
  <div
    className="w-full md:w-1/2 px-8"
  >
    <Connect(PostList) />
  </div>
  <div
    className="w-full md:w-1/2 px-8"
  >
    <Connect(ActionList) />
  </div>
</div>
`;
